v1/openings/views/opening.py

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+from rest_framework import status
+from rest_framework.mixins import (
+    CreateModelMixin,
+    DestroyModelMixin,
+    ListModelMixin,
+    RetrieveModelMixin,
+    UpdateModelMixin
+)
+from rest_framework.response import Response
+from rest_framework.viewsets import GenericViewSet
+
+from v1.third_party.rest_framework.permissions import IsStaffOrReadOnly
+from ..models import Opening
+from ..serializers import OpeningSerializer, OpeningSerializerWrite
+
+
+class OpeningViewSet(
+    CreateModelMixin,
+    DestroyModelMixin,
+    ListModelMixin,
+    RetrieveModelMixin,
+    UpdateModelMixin,
+    GenericViewSet,
+):
+    queryset = Opening.objects \
+        .prefetch_related('reports_to', 'responsibilities', 'skills') \
+        .order_by('created_date') \
+        .all()
+    serializer_class = OpeningSerializer
+    serializer_write_class = OpeningSerializerWrite
+    pagination_class = None
+    permission_classes = [IsStaffOrReadOnly]
+
+    def create(self, request, *args, **kwargs):
```

Comment:
in case when you want to supply different serializers to create and update methods its better to redefine 
```
    def get_serializer_class(self):
        if self.action == 'create':
            return CreateSerializer
        return DefaulSerializer 
```

Reply:
and you won't need to redefine create and update methods

Reply:
I was attempting to allow the request and response to be different shapes (nested data serialized in response). When I used this method both the request and response were always the same shape (only the `pk` of the related objects in the response).